File: packages/jupyterlab-kuusi/src/releaseTier.ts
import { KUUSI_VERSION } from "./version";

export type KuusiReleaseTier = "core" | "full";

const RELEASE_TIER_OVERRIDE_KEY = "jupyterlab-kuusi:release-tier-override";

const tierFromVersion = (version: string): KuusiReleaseTier => {
  const minor = Number.parseInt(version.split(".")[1] ?? "0", 10);
  return Number.isFinite(minor) && minor >= 3 ? "full" : "core";
};

/** Default tier from semver: 0.2.x → core, 0.3+ → full. */
export const KUUSI_RELEASE_TIER: KuusiReleaseTier = tierFromVersion(KUUSI_VERSION);

const readTierOverride = (): KuusiReleaseTier | null => {
  try {
    const raw = localStorage.getItem(RELEASE_TIER_OVERRIDE_KEY);

    if (raw === "core" || raw === "full") {
      return raw;
    }
  } catch {
    // private browsing / restricted storage
  }

  return null;
};

export const getKuusiReleaseTier = (): KuusiReleaseTier =>
  readTierOverride() ?? KUUSI_RELEASE_TIER;

export const isKuusiFullRelease = (): boolean =>
  getKuusiReleaseTier() === "full";

export const isKuusiCoreRelease = (): boolean =>
  getKuusiReleaseTier() === "core";

File: packages/jupyterlab-kuusi/jupyterlab_kuusi/release_tier.py
"""Release tier: 0.2.x ships core (mind map + settings); 0.3+ enables full Kuusi."""

from __future__ import annotations

import os

from ._version import __version__


def _tier_from_version(version: str) -> str:
    parts = version.split(".")
    try:
        minor = int(parts[1]) if len(parts) > 1 else 0
    except ValueError:
        minor = 0
    return "full" if minor >= 3 else "core"


def get_release_tier() -> str:
    override = os.environ.get("KUUSI_RELEASE_TIER", "").strip().lower()
    if override in ("core", "full"):
        return override
    return _tier_from_version(__version__)


def is_full_release() -> bool:
    return get_release_tier() == "full"
